runtime/compiler/control/JITClientCompilationThread.cpp

```diff
@@ -3108,9 +3119,13 @@ remoteCompile(
    std::pair<std::string, std::string> chtableUpdates = table->serializeUpdates();
    // Update the sequence number for these updates
    uint32_t seqNo = compInfo->incCompReqSeqNo();
-   uint32_t lastCriticalSeqNo = compInfo->getLastCriticalSeqNo();
+   uint32_t lastCriticalSeqNo = !details.isJitDumpMethod() ? compInfo->getLastCriticalSeqNo() : seqNo;
```

Comment:
For a diagnostic compilation, should we set `lastCriticalSeqNo` to `seqNo` or to 0?
At the server, the current request is allowed to pass only if the critical seqno that this request depends on has already passed. We know that the request with seqNo==0 has already passed, so I am inclined to use 0 here.

Reply:
I saw the change in JITServerCompilationThread.cpp processEntry() that allows the diagnostic compilations to pass. If we force our way though, maybe it doesn't matter the value of `lastCriticalSeqNo` is.

Reply:
Actually, setting to 0 on the client I think is better, since this means we don't have to hack the server as well.